### ISSUE TITLE
[question] Does coursier support resolve by file path?

### ISSUE THREAD

**Issue by wisechengyi** [coursier/coursier#710]
Basically resolve an artifact based on a relative or absolution file path. For example:

https://stackoverflow.com/questions/2229757/maven-add-a-dependency-to-a-jar-by-relative-path

so in some cases it can be useful for user who wants to resolve adhoc jars.

**Comment by alexarchambault** [comment]
@wisechengyi One can always add file-based repositories, like `coursier fetch -r file:///path/to/repo …`

**Comment by wisechengyi** [comment]
How do you feel about if I add the feature to take a single jar file path that is not in any repo (with no pom, so intransitivity implied)? Basically it allows user to consume a single jar without having to create a local maven repo.

**Comment by wisechengyi** [comment]
@alexarchambault ^

**Comment by alexarchambault** [comment]
@wisechengyi I'm not sure I understand what you mean… I've been willing to add something like a `download` command to the cli, like
```
$ coursier download https://foo.com/something
```
to download and cache the passed url, relying on the cache / ttl / etc. of coursier. Is it what you mean?

**Comment by wisechengyi** [comment]
Hi @alexarchambault, sorry about the delay. Yes I think so, although I was thinking adding it directly to `fetch` command, so it would be something like:
```
coursier fetch --intransitive a:b:c:url=file://.../x.jar
```
and then use coursier's cache ttl to manage. Say `x.jar` changes within the TTL, then coursier should not fetch from the path again.

**Comment by alexarchambault** [comment]
You mean, the supplied file would act as a dependency-free dependency, with only the supplied file as artifact? It's not the exact use case I was thinking of, but I'm not against it either!

**Comment by wisechengyi** [comment]
Correct. Cool thanks.

**Comment by alexarchambault** [comment]
FYI There are already [related things](https://github.com/coursier/coursier/blob/a41c07194344555f5f9ee7a6e5b3c01ff13afa7b/sbt-coursier/src/main/scala/coursier/FallbackDependenciesRepository.scala) for the sbt plugin (to handle things like `libraryDependencies += "org" % "name" % "ver" from "url"`)

**Comment by wisechengyi** [comment]
Yeah I think that's the same idea, so maybe just need to plumb it for cli then.

**Comment by wisechengyi** [comment]
@alexarchambault the syntax I'd like to propose is `org:name:version,classifier=<classifier>,url=<encoded url file://... or https://...>` by leveraging #735

**Comment by wisechengyi** [comment]
Resolved by #774
